i2c: stm32: add missing DMA configuration fields

sgilbert182 opened the pull request:
Add missing fields for DMA tx and rx configuration macros

fix #87516

Tested on stm32h735-Discovery and STM32H747i-Discovery board

sgilbert182 committed:
i2c: stm32: add missing DMA configuration fields

Add missing fields for DMA tx and rx configuration macros

Signed-off-by: Simon Gilbert <srdgilbert@gmail.com>

erian747 commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -587,15 +587,56 @@ static void i2c_stm32_irq_config_func_##index(const struct device *dev)	\
+void i2c_stm32_dma_rx_cb(const struct device *dma_dev, void *user_data,
+	uint32_t channel, int status)
+{
+	struct i2c_stm32_data *data = user_data;
```
Compiler warning about unused variable

sgilbert182 replied:
Removed variable

erian747 commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -587,15 +587,56 @@ static void i2c_stm32_irq_config_func_##index(const struct device *dev)	\
+void i2c_stm32_dma_tx_cb(const struct device *dma_dev, void *user_data,
+	uint32_t channel, int status)
+{
+	struct i2c_stm32_data *data = user_data;
```
Compiler warning about unused variable

sgilbert182 commented on drivers/i2c/i2c_ll_stm32_v2.c:
```diff
@@ -195,10 +195,13 @@ static inline void msg_init(const struct device *dev, struct i2c_msg *msg,
 				data->dma_blk_cfg.source_address = LL_I2C_DMA_GetRegAddr(
 					cfg->i2c, LL_I2C_DMA_REG_DATA_RECEIVE);
+				data->dma_blk_cfg.source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+
```
Done

sgilbert182 commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -587,15 +587,52 @@ static void i2c_stm32_irq_config_func_##index(const struct device *dev)	\
+		},
+
+void i2c_stm32_dma_tx_cb(const struct device *dma_dev, void *user_data,
+	uint32_t channel, int status)
```
Done

sgilbert182 commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -587,15 +587,52 @@ static void i2c_stm32_irq_config_func_##index(const struct device *dev)	\
+	if (status != 0) {
+		LOG_ERR("DMA error %d", status);
+	}
+}
```
Tidied up the functions and broken out into it's own commit with an explanation for their use

sgilbert182 committed:
i2c: stm32: add DMA callback stubs to avoid nullptr calls in ISR context

Add stub functions for the I2C DMA callbacks, which are invoked
during or upon completion of DMA-based I2C transactions. Without
these, NULL pointer calls occur on DMA transfer complete or error
events, leading to faults within ISR context.

Signed-off-by: Simon Gilbert <srdgilbert@gmail.com>

erwango commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -595,6 +595,32 @@ static void i2c_stm32_irq_config_func_##index(const struct device *dev)	\
+void i2c_stm32_dma_tx_cb(const struct device *dma_dev, void *user_data,
+			uint32_t channel, int status)
+{
+	(void)dma_dev;
+	(void)user_data;
+	(void)channel;
```
```suggestion
	ARG_UNUSED(dma_dev);
	ARG_UNUSED(user_data);
	ARG_UNUSED(channel);
```

sgilbert182 replied:
Done

sgilbert182 commented:
Hi @erwango, this seems to have sat stale for a while but is tested working and is good to go. Are there any more changes you need?

gautierg-st commented on drivers/i2c/i2c_ll_stm32.h:
```diff
@@ -105,7 +105,8 @@ struct i2c_stm32_data {
 #endif
 #ifdef CONFIG_I2C_STM32_V2_DMA
-	struct dma_config dma_cfg;
+	struct dma_config dma_tx_cfg;
+	struct dma_config dma_rx_cfg;
```
Nit (non-blocking): since struct `i2c_stm32_config` has `tx_dma` and `rx_dma`, it would be more logical to name these `tx_dma_cfg` and `rx_dma_cfg` (or the other way around by renaming the first two as `dma_tx` and `dma_rx`)
